packages/react-native-aztec/src/AztecInputState.js

```diff
@@ -71,15 +71,18 @@ export const getCurrentFocusedElement = () => {
 export const notifyInputChange = () => {
 	const focusedInput = TextInputState.currentlyFocusedInput();
 	const hasAnyFocusedInput = focusedInput !== null;
+	const hasListeners = focusChangeListeners.length > 0;
 
-	if ( hasAnyFocusedInput ) {
-		if ( ! currentFocusedElement ) {
-			notifyListeners( { isFocused: true } );
+	if ( hasListeners ) {
```

Comment:
I'm currently having issues with `notifyInputChange` when opening the editor without any content, just the default Paragraph block 😅 

When I tap on the block the default state within `BlockDraggable` for `isEditingText` is set to `false`, then the `focus` event is triggered which calls `notifyInputChange` at this point the listener is not yet added, it happens after it hence the state of `isEditingText` never gets updated because it's already in focus and `notifyListeners` doesn't get triggered again.

Maybe we could update the following:

`const focusedInput = TextInputState.currentlyFocusedInput();`

To be: 

`	const focusedInput =
		TextInputState.currentlyFocusedInput() &&
		currentFocusedElement !== null;`

What do you think?

Reply:
That's interesting 🤔. I've been testing locally and found that the culprit is related to the initial value of `isEditingText` state value.

In order to debug the focus changes and the initial state value, I added some console logs within the code. I obtained the following logs after creating a new post and tapping on the Paragraph block:
```
BlockDraggable - isEditingText - initial value: false
AztecInputState - focus()
AztecInputState - notifyListeners {"isFocused": true, "listeners": 0}
BlockDraggable - Add listener 65b9a13b-f6af-48fd-a629-b3c0c56ea807
AztecInputState - focus()
```

As you can see, the initial value of `isEditingText` is false because on the first render call the `focus` function of the `AztecView` component hasn't been called yet. When `BlockDraggable` adds the listener, it's too late because the `focus` event happened before, and the next `focus` event won't be considered as a focus change. This leads to the Paragraph block considering that the user is not editing text and hence, allowing the dragging gesture.

This race condition can be easily fixed by initializing `isEditingText` at the same time the focus change listener is added:
```diff
diff --git a/packages/block-editor/src/components/block-draggable/index.native.js b/packages/block-editor/src/components/block-draggable/index.native.js
index f291a20edb2f145f0d591eea83f3b38dfb0ae4dd..6d127782980f8efec26919b2bd15eb65424f8ecd 100644
--- a/packages/block-editor/src/components/block-draggable/index.native.js
+++ b/packages/block-editor/src/components/block-draggable/index.native.js
@@ -262,9 +262,7 @@ const BlockDraggableWrapper = ( { children } ) => {
  */
 const BlockDraggable = ( { clientId, children, enabled = true } ) => {
 	const wasBeingDragged = useRef( false );
-	const [ isEditingText, setIsEditingText ] = useState(
-		RCTAztecView.InputState.isFocused()
-	);
+	const [ isEditingText, setIsEditingText ] = useState( false );
 
 	const draggingAnimation = {
 		opacity: useSharedValue( 1 ),
@@ -327,6 +325,11 @@ const BlockDraggable = ( { clientId, children, enabled = true } ) => {
 	}, [] );
 
 	useEffect( () => {
+		const isAnyAztecInputFocused = RCTAztecView.InputState.isFocused();
+		if ( isAnyAztecInputFocused ) {
+			setIsEditingText( isAnyAztecInputFocused );
+		}
+
 		RCTAztecView.InputState.addFocusChangeListener( onFocusChangeAztec );
 		return () => {
 			RCTAztecView.InputState.removeFocusChangeListener(
```

As far as I tested locally, this will address the original issue, and won't be necessary to add the `hasListeners` condition here. @geriux let me know your opinion on this workaround, I'll be happy to help out in the implementation.

Reply:
Thank you for investigating and providing a workaround for this issue! I'm testing it's working well, I'll update the code accordingly to remove `hasListeners` and include this.

Reply:
I applied the above in https://github.com/WordPress/gutenberg/pull/40788/commits/e3a82e9bd66e59f52cbf3a04460969afdb802597